packages/react/src/ReactElementValidator.js

```diff
@@ -213,30 +225,39 @@ function validateChildKeys(node, parentType) {
  * @param {ReactElement} element
  */
 function validatePropTypes(element) {
-  const componentClass = element.type;
-  if (typeof componentClass !== 'function') {
+  const type = element.type;
+  let name, propTypes;
```

Comment:
Shouldn't it be `const propTypes = type.propTypes` after the `typeof type` check, as that's the only value that it gets assigned?

Reply:
The current approach avoids reading `propTypes` from a possible `null` or `undefined` value (although that shouldn't actually happen in practice). I don't think it's a problem as-is.